repo-agent/pkg/llm/claude.go

```diff
@@ -66,36 +56,21 @@ func (c *Claude) AddPostProcessor(p PostProcessor) {
 	c.postProcessors = append(c.postProcessors, p)
 }
 
-func (c *Claude) Setup(_, _ string) error {
-	// First, try to get the config from in-cluster config
-	config, err := inClusterConfigFunc()
+func (c *Claude) Setup(_, tokensDir string) error {
+	// Read API key from the mounted secret file
+	apiKeyPath := filepath.Join(tokensDir, AnthropicAPIKeySecretKey)
+	apiKeyBytes, err := os.ReadFile(apiKeyPath)
 	if err != nil {
-		// If that fails, fall back to checking the environment variable
-		log.Printf("Not running in a Kubernetes cluster, checking for ANTHROPIC_API_KEY environment variable: %v", err)
+		// If reading from file fails, fall back to checking the environment variable
+		log.Printf("Failed to read API key from %s: %v", apiKeyPath, err)
 		apiKey, ok := os.LookupEnv(AnthropicAPIKeyEnvVar)
```

Comment:
The function signature `Setup(_, tokensDir string)` has an unused first argument. Consider removing it if it's not needed by the interface.